cpg_workflows/stages/aip.py

```diff
@@ -0,0 +1,294 @@
+
+CHUNKY_DATE = get_config()['workflow'].get('fake_date', datetime.now().strftime('%Y-%m-%d'))
+DATED_FOLDER = join('reanalysis', CHUNKY_DATE)
+
+
+@stage
+class GeneratePanelData(DatasetStage):
+    """
+    PythonJob to find HPO-matched panels
+    """
+
+    def expected_outputs(self, dataset: Dataset) -> dict:
+        """
+        only one output, the panel data
+        """
+        return {'hpo_panels': dataset.prefix() / DATED_FOLDER / 'hpo_panel_data.json'}
+
+    def queue_jobs(self, dataset: Dataset, inputs: StageInput) -> StageOutput | None:
+        py_job = get_batch().new_python_job('create_panel_data')
+        py_job.image(get_config()['workflow']['driver_image'])
+
+        # only needs a really tiny image
+        STANDARD.set_resources(py_job, ncpu=1, mem_gb=1.0)
+        expected_d = self.expected_outputs(dataset)
+        hpo_file = get_batch().read_input(get_config()['workflow']['obo_file'])
+        py_job.call(
+            panel_match_main, dataset.name, hpo_file, str(expected_d['hpo_panels'])
+        )
+
+        return self.make_outputs(dataset, data=expected_d, jobs=py_job)
+
+
+@stage(required_stages=[GeneratePanelData])
+class QueryPanelapp(DatasetStage):
+    """
+    query PanelApp for up-to-date gene lists
+    """
+
+    def expected_outputs(self, dataset: Dataset) -> dict[str, Path]:
+        return {'panel_data': dataset.prefix() / DATED_FOLDER / 'panelapp_data.json'}
+
+    def queue_jobs(self, dataset: Dataset, inputs: StageInput) -> StageOutput:
+
+        job = get_batch().new_job('query panel data')
+        STANDARD.set_resources(job, ncpu=1, mem_gb=1.0)
+        job.image(image_path('aip'))
+
+        # auth and copy env
+        authenticate_cloud_credentials_in_job(job)
+        copy_common_env(job)
+
+        hpo_panel_json = inputs.as_path(
+            target=dataset, stage=GeneratePanelData, key='hpo_panels'
+        )
+        expected_out = self.expected_outputs(dataset)
+        job.command(
+            f'python3 reanalysis/query_panelapp.py '
+            f'--panels {str(hpo_panel_json)} '
+            f'--out_path {str(expected_out["panel_data"])}'
+        )
+
+        return self.make_outputs(dataset, data=expected_out, jobs=job)
+
+
+def query_for_latest_mt(dataset: str) -> str:
+    """
+    query for the latest MT for a dataset
+    Args:
+        dataset (str):
+    Returns:
+        str, the path to the latest MT
+    """
+    my_query = gql(
+        """
+        query MyQuery($dataset: String!) {
+            project(name: $dataset) {
+                analyses(type: {eq: "custom"}, status: {eq: COMPLETED}) {
+                    output
+                }
+            }
+        }
+    """
+    )
+    result = query(my_query, variables={'dataset': dataset})
+    seq_type_exome = get_config()['workflow'].get('sequencing_type') == 'exome'
+    mt_path = ''
+    for analysis in result['project']['analyses']:
+        if analysis['output'].endswith('.mt') and (
```

Comment:
An analysis entry can have None for an `analysis['output']` - allow for this instead of failing to do .endswith on a None

Reply:
it would be good to get a specific analysis type for MTs, at the moment they're all logged as `custom`. If there's an MT type the output would never be empty